### PR TITLE
Fix number of threads for pushing to views

### LINKED ISSUE #37900: max_insert_threads and materialized views performance degradation
test
```sql
DROP TABLE IF EXISTS t;
DROP TABLE IF EXISTS t_mv;
create table t (a UInt64) Engine = Null;
create materialized view t_mv Engine = Null AS select now() as ts, max(a) from t group by ts;
insert into t select * from numbers_mt(3000000000) settings max_threads = 16, max_insert_threads=16;
```

results:
```
21.8:  
0 rows in set. Elapsed: 3.841 sec. Processed 3.00 billion rows, 24.00 GB (781.01 million rows/s., 6.25 GB/s.)

21.10: 
0 rows in set. Elapsed: 3.844 sec. Processed 6.00 billion rows, 48.00 GB (1.56 billion rows/s., 12.49 GB/s.)

21.11:
0 rows in set. Elapsed: 17.717 sec. Processed 5.98 billion rows, 47.86 GB (337.70 million rows/s., 2.70 GB/s.)

21.12
0 rows in set. Elapsed: 18.569 sec. Processed 6.00 billion rows, 48.00 GB (323.12 million rows/s., 2.58 GB/s.)

22.1
0 rows in set. Elapsed: 19.124 sec. Processed 6.00 billion rows, 48.00 GB (313.74 million rows/s., 2.51 GB/s.)

22.2
0 rows in set. Elapsed: 19.939 sec. Processed 6.00 billion rows, 48.00 GB (300.92 million rows/s., 2.41 GB/s.)

22.3
0 rows in set. Elapsed: 18.727 sec. Processed 6.00 billion rows, 48.00 GB (320.39 million rows/s., 2.56 GB/s.)

22.4
0 rows in set. Elapsed: 19.698 sec. Processed 6.00 billion rows, 48.00 GB (304.60 million rows/s., 2.44 GB/s.)

22.5
0 rows in set. Elapsed: 22.723 sec. Processed 6.00 billion rows, 48.00 GB (264.05 million rows/s., 2.11 GB/s.)
```

SIde note: pay attention to processed rows and progress bar behavior (minor cosmetic issue).

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `parallel_view_processing=1` with `optimize_trivial_insert_select=1`. Fix `max_insert_threads` while pushing to views.

Refs: #12085
Refs: #29786
Refs: #28582
Fixes: #37900 (cc @filimonov )
Cc: @KochetovNicolai 

_Note: these two fixes in different patches._

### CODE REVIEW

File: src/Interpreters/InterpreterInsertQuery.cpp [hunk 1]
```diff
@@ -457,20 +457,26 @@ BlockIO InterpreterInsertQuery::execute()
...
 
         pipeline.addChains(std::move(out_chains));
 
-        pipeline.setMaxThreads(num_insert_threads);
-        /// Don't use more threads for insert then for select to reduce memory consumption.
-        if (!settings.parallel_view_processing && pipeline.getNumThreads() > num_select_threads)
-            pipeline.setMaxThreads(num_select_threads);
+        if (!settings.parallel_view_processing)
+        {
+            /// Don't use more threads for insert then for select to reduce memory consumption.
```
Comment (alexey-milovidov) [inline src/Interpreters/InterpreterInsertQuery.cpp:468]: then -> than

**Review by alexey-milovidov — APPROVED** [review]

**Comment by azat** [comment]
>[Performance Comparison [4/4] — Failed to parse the report.](https://s3.amazonaws.com/clickhouse-test-reports/38731/b3c0e396ed41cb9aaadd30590f1319e2d71e91f8/performance_comparison_[4/4]/runlog.log)

```
output/clickhouse
tar: clickhouse: Wrote only 3072 of 10240 bytes
2022-07-03 11:14:10 URL:https://s3.amazonaws.com/clickhouse-builds/38731/b3c0e396ed41cb9aaadd30590f1319e2d71e91f8/package_release/performance.tgz [656960872/656960872] -> "-" [1]
output/clickhouse-copier
output/clickhouse-disks
tar: Exiting with failure status due to previous errors
```

**Comment by azat** [comment]
>[Performance Comparison [3/4] — 2 too long, 5 faster, 1 slower, 1 unstable](https://s3.amazonaws.com/clickhouse-test-reports/38731/7427adb600633008af5688e394eb287f370e1d41/performance_comparison_[3/4]/report.html)

Old, s | New, s | Ratio of speedup (-) or slowdown (+) | Relative difference (new − old) / old | p < 0.01 threshold | Test | # | Query
-- | -- | -- | -- | -- | -- | -- | --
6.894 | 0.825 | -8.356x | -0.881 | 0.880 | views_max_insert_threads | 0 | `insert into views_max_insert_threads_null select * from numbers_mt(3000000000) settings max_threads = 16, max_insert_threads=16`

**Comment by azat** [comment]
>[Stress test (memory, actions) — Backward compatibility check: Error message in clickhouse-server.log (see bc_check_error_messages.txt)](https://s3.amazonaws.com/clickhouse-test-reports/38731/7427adb600633008af5688e394eb287f370e1d41/stress_test__memory__actions_.html)

```
2022.07.03 18:02:45.335507 [ 388363 ] {} <Error> test_29.tp_1 (07040515-bc40-4602-adf2-83c9d6e37a37): Cannot quickly remove directory /var/lib/clickhouse/store/070/07040515-bc40-4602-adf2-83c9d6e37a37/delete_tmp_all_0_0_0_2 by removing files; fallback to recursive removal. Reason: Code: 458. DB::ErrnoException: Cannot unlink file /var/lib/clickhouse/store/070/07040515-bc40-4602-adf2-83c9d6e37a37/delete_tmp_all_0_0_0_2/pp.proj, errno: 21, strerror: Is a directory. (CANNOT_UNLINK) (version 22.7.1.1)
```

Projections cannot be removed quickly, but this will go away after #38717

**Pull request merged**